docs/source/contributor-guide/benchmarking_macos.md

```diff
@@ -0,0 +1,145 @@
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# Comet Benchmarking on macOS
+
+This guide is for setting up TPC-H benchmarks locally on macOS using the 100 GB dataset.
+
+Note that running this benchmark on macOS is not ideal because we cannot force Spark or Comet to use performance 
+cores rather than efficiency cores, and background processes are sharing these cores. Also, power and thermal 
+management may throttle CPU cores.  
+
+## Prerequisites
+
+Java and Rust must be installed locally.
+
+## Data Generation
+
+```shell
+cargo install tpchgen-rs
+tpchgen-cli -s 100 --format=parquet
+```
+
+## Clone the DataFusion Benchmarks Repository
+
+```shell
+git clone https://github.com/apache/datafusion-benchmarks.git
+```
+
+## Install Spark
+
+Install Spark
+
+```shell
+wget https://archive.apache.org/dist/spark/spark-3.5.4/spark-3.5.4-bin-hadoop3.tgz
+tar xzf spark-3.5.4-bin-hadoop3.tgz
+sudo mv spark-3.5.4-bin-hadoop3 /opt
+export SPARK_HOME=/opt/spark-3.5.4-bin-hadoop3/
+mkdir /tmp/spark-events
+```
+
+
+Start Spark in standalone mode:
+
+```shell
+$SPARK_HOME/sbin/start-master.sh
+```
+
+Set `SPARK_MASTER` env var (host name will need to be edited):
+
+```shell
+export SPARK_MASTER=spark://Rustys-MacBook-Pro.local:7077
```

Comment:
localhost?

Reply:
Spark master does not bind to `localhost` by default. We could specify `--host localhost` when starting the master process, but I have not tried this on macOS yet. I will test this and update this PR.

Reply:
FWIW, this is what I currently use to start Spark for benchmarking TPC-H on macOS:
```
export SPARK_HOME=/opt/spark-3.5.5-bin-hadoop3
export COMET_JAR=/Users/matt/git/datafusion-comet/spark/target/comet-spark-spark3.5_2.12-0.8.0-SNAPSHOT.jar
export SPARK_MASTER="local[*]"
export SPARK_MASTER_HOST="127.0.0.1"
export SPARK_LOCAL_IP="127.0.0.1"

mkdir /tmp/spark-events

$SPARK_HOME/sbin/start-master.sh
$SPARK_HOME/sbin/start-worker.sh $SPARK_MASTER_HOST:7077
$SPARK_HOME/sbin/start-history-server.sh
```

Reply:
I have been using standalone mode rather than local mode, but perhaps local mode may make more sense for macOS.